src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/ActivityResourceIntTest.kt

```diff
@@ -681,24 +676,26 @@ class ActivityResourceIntTest : IntegrationTestBase() {
 
       @Test
       fun `should return OK for empty pay rates`() {
-        val existingActivityId = getSavedActivityId()
-
         callUpdateEndpoint(
-          courseActivityId = existingActivityId,
+          courseActivityId = courseActivity.courseActivityId,
           jsonBody = updateActivityRequestJson(payRatesJson = """ "payRates" : [],"""),
         )
           .expectStatus().isOk
 
-        val updated = repository.lookupActivity(existingActivityId)
+        val updated = repository.lookupActivity(courseActivity.courseActivityId)
         assertThat(updated.payRates[0].endDate).isEqualTo(LocalDate.now())
       }
 
       @Test
       fun `should return bad request if pay rate removed which is allocated to an offender`() {
         testData(repository) {
-          courseActivity = courseActivity {
-            payRate(iepLevelCode = "STD")
-            payRate(iepLevelCode = "BAS")
+          programService {
+            courseActivity = courseActivity {
+              payRate(iepLevelCode = "STD")
+              payRate(iepLevelCode = "BAS")
+              courseSchedule()
+              courseScheduleRule()
+            }
```

Comment:
This is the point of the PR - filled out the entity hierarchy from program service -> course activity -> children. 
One thing I'm not sure of is that when using builders only we used to default a single pay rate, course schedule and course schedule rule which could all be overridden with an empty list or a list full of stuff as the test dictated. Now we default an empty list due to the DSL pattern we need to specify those single defaults - can't decide whether to go back to defauling them again or keeping their declaration explicit 🤔